app/bot/when.go

```diff
@@ -0,0 +1,134 @@
+package bot
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+)
+
+// When bot is answer on question "when the stream is started".
+type When struct {
+	now     func() time.Time
+	reactOn []string
+}
+
+// NewWhen makes a new When bot.
+func NewWhen(reactOn []string, nowFunc func() time.Time) *When {
+	log.Printf("[INFO] new when bot is started")
+
+	return &When{
+		reactOn: reactOn,
+		now:     nowFunc,
+	}
+}
+
+// Help returns help message
+func (w *When) Help() string {
+	return genHelpMsg(w.ReactOn(), "расписание эфиров Радио-Т")
+}
+
+// OnMessage returns one entry
+func (w *When) OnMessage(msg Message) Response {
+	if !contains(w.ReactOn(), msg.Text) {
+		return Response{}
+	}
+
+	return Response{
+		Text: when(w.now()),
+		Send: true,
+	}
+}
+
+// ReactOn keys
+func (w *When) ReactOn() []string {
+	return w.reactOn
+}
+
+func when(now time.Time) string {
+	const avgDuration = 2 * time.Hour
+	const whenPrefix = "[каждую субботу, 20:00 UTC](https://radio-t.com/online/)"
+	const tooSmallDuration = "пару секунд"
+
+	now = now.UTC()
+	prevStream, nextStream := closestPrevNextWeekdays(now, time.Saturday, 20, 0)
+	diffToPrev := -prevStream.Sub(now)
+	diffToNext := nextStream.Sub(now)
+
+	var whenCountdown string
+	switch {
+	case diffToPrev < avgDuration:
+		whenCountdown = fmt.Sprintf(
+			"\nНачался %s назад. \nСкорее всего еще идет. \nСледующий через %s",
+			humanizeDuration(diffToPrev, tooSmallDuration),
+			humanizeDuration(diffToNext, tooSmallDuration),
+		)
+	case diffToNext >= 0:
+		whenCountdown = fmt.Sprintf(
+			"\nНачнется через %s",
+			humanizeDuration(diffToNext, tooSmallDuration),
+		)
+	}
+
+	return whenPrefix + whenCountdown
+}
+
+// closestPrevNextWeekdays returns closest next `weekday` at `hour`:`minute` after `t`.
+func closestPrevNextWeekdays(t time.Time, weekday time.Weekday, hour, minute int) (time.Time, time.Time) {
+	const daysInWeek = 7
+	const week = daysInWeek * 24 * time.Hour
+
+	daysDiff := int((daysInWeek + (weekday - t.Weekday())) % daysInWeek)
+	year, month, day := t.AddDate(0, 0, daysDiff).Date()
+
+	nextDt := time.Date(year, month, day, hour, minute, 0, 0, t.Location())
+	if t.After(nextDt) {
+		return nextDt, nextDt.Add(week)
+	}
+
+	return nextDt.Add(-week), nextDt
+}
+
+func humanizeDuration(d time.Duration, defaultVal string) string {
```

Comment:
probably adding https://github.com/dustin/go-humanize will be simpler and more generic solution here.

Reply:
This library have a little bit strange api. I will integrate this, but I don't like this solution. Especially because the output is contains only biggest dimension in result. 

for duration `1d23h58m59s` current solution respond with `1d 23h`, but humanize returns `1 day`. So, it cut-off more than 1 first dimension. In out case it is not good. Am I right?

Anyway - plz see a [commit with humanize](https://github.com/radio-t/super-bot/pull/83/commits/1d2d8d74142ae50d0c6e6e621ebe63130fcf76e6).

Reply:
🤦🏻 I found `HumanizeDuration` in `bot/humanize.go`. What we should do with that?)

Reply:
To be clear: I do not like humanize impl and want to rollback changes

Reply:
Well, I forgot we have `HumanizeDuration` already in place. Looks like it is used by wtf bot already, so to me, it makes sense just to use it in this new bot as is

Reply:
fixed